glom/trace.py

```diff
@@ -0,0 +1,125 @@
+"""
+this module contains helpers for building glom data
+structure stack traces from glom scopes
+
+there are a few cursors that are extracted from each level of the
+scope; these cursors say which path down from a target and spec
+was being processed
+
+scope[Path] -- target cursor
+scope[SCOPE_POS] -- spec cursor (for dict + tuple type specs)
+"""
+from __future__ import print_function
+
+from .core import Spec, T, UP, Path, TType
+
+
+_NO_TARGET = object()
+
+
+def _unpack_stack(scope):
+    """
+    convert scope to [(scope, spec, target)]
+    """
+    # root glom call generates two scopes up at the top that aren't part
+    # of execution
+    return [(scope, scope[Spec], scope[T]) for scope in list(reversed(scope.maps[:-2]))]
+
+
+# TODO: get fancier here and replace repr()
+# with something that only recurses one level
+def _format_value(value, maxlen):
+    s = repr(value)
+    if len(s) > maxlen:
+        s = s[:maxlen] + '...'
```

Comment:
Agree we should do something fancier, but for now I'd do `max(maxlen - 3, 0)` at least

Reply:
oh - 3 is a good idea